### ISSUE TITLE
Problema 8

### ISSUE THREAD

**Issue by AsLogd** [AsLogd/Problemes-Ampliaci-Algorismia#4]
Arnau:
Estoy haciendo la segunda parte del problema 8 y no tengo claro como define el problema. Dice que contemos los paths entre dos vértices y que en estos paths pueden haber nodos repetidos. No tengo claro si también se pueden repetir edges (ya que no lo prohibe explicitamente) y tampoco si pueden salir repetidos los nodos de inicio y destino en alguno de los nodos intermedios del path. Vosotros qué entendéis?

**Comment by AsLogd** [comment]
![imagen](https://user-images.githubusercontent.com/6876725/41192761-0efd70f2-6c03-11e8-92df-ffe1b2f7b7dc.png)
Define el numero de caminos entre s y t como una tupla de tamaño |V|-1 de la forma (v_0,...,v_k) (una tupla que ordena los vertices por los que se pasan), donde cada vertice tiene que tener una arista al contiguo (logico). Si hubiera (en un mismo camino) una arista repetida, tendrias un ciclo, y podrias formar un camino recortando ese ciclo. En principio no veo que haya problema en que se repita el nodo de inicio, pero el de final seria raro, no? le dare un par de vueltas mas

**Comment by gonsp** [comment]
Puesto que no lo especifican diría que sí que puedes repetir nodos y aristas (incluyendo a s y t)

**Comment by arnaubadia** [comment]
Le mandé un email a la profesora para que me lo confirmase. Me dijo esto:

"Hola, Arnau.
Els camins poden repetir vèrtexs (incloent s i t) i també arcs. No imposo cap restricció.  Així l'algorisme pel problema de comptar camins veuràs que és molt més senzill.
Salutacions,
Carme"

Llevaba horas dando vueltas al problema considerando que no se podian repetir edges y no encontraba solución. Pero como dice Carme si se pueden repetir edges el problema es bastante sencillo, usando Dynamic Programming.

**Comment by gonsp** [comment]
Sep, con dynamic programming se resuelve. Personalmente ya lo tengo claro así que no creo que haga falta gastar tiempo en pasarlo en latex, si acaso podemos escribir la fórmula recursiva en un papel y la pasamos por aquí

**Comment by arnaubadia** [comment]
Estoy de acuerdo con que no hace falta lo del latex. A qué te refieres exactamente con fórmula recursiva? Yo había pensado en dar el algoritmo de cómo construir la tabla. Y luego haría falta demostrar por qué es correcto.

**Comment by AsLogd** [comment]
Suponho que se refiere a describir cuales son los subproblemas a resolver

**Comment by gonsp** [comment]
Osea es simplemente definir la función:
![image](https://user-images.githubusercontent.com/11088802/41205668-fbe8f05e-6cf7-11e8-8e29-183e0f85ff04.png)

**Comment by gonsp** [comment]
Los otros apartados, sería argumentar que está en FSPACE porque podemos hacer un backtracking escogiendo todos los valores posibles de las variables y incrementar un contador si satisface la fórmula (espacio polinómico), y decir que es NP-Hard porque podemos reducir SAT a este.
No?

**Comment by ghost** [comment]
@GonMolon Sí, comprobando si hay 1 o más asignaciones.
En cuanto al code, yo he usado una matriz D[u][d], donde significa: "Número de caminos de s a u con como mucho distancia d" y se devuelve D[t][d-1]. En tu caso al usar llamadas recursivas quizás tienes que justificar que no es EXP.

**Comment by gonsp** [comment]
Si bueno, también almaceno los resultados de cada llamada en una tabla. Es lo mismo. Aunque voy a pensar sobre lo de en vez de preguntar cuántos caminos distintos hay con k pasos, preguntar por k o menos pasos

**Comment by Urigueler** [comment]
A mi la solució em dona polinòmica en relació al nombre de camins...Heu aconseguit ferho polinòmic en relació a l'entrada? (de l'últim apratat)

**Comment by gonsp** [comment]
mmm la solución de dynamic programming es O(|V| * |E| * k) creo

**Comment by DavidMoranPomes** [comment]
A mi me sale un dynamic programmin con coste O(|V|^2 * |E|)

**Comment by gonsp** [comment]
Bueno es lo mismo, porque k <= |V|

**Comment by DavidMoranPomes** [comment]
aaaah vale, pensaba que vuestra k era el número de caminos. Es lo mismo entonces.

**Comment by DavidMoranPomes** [comment]
Os dejo igualmente mi [formalización del problema](https://github.com/AsLogd/Problemes-Ampliaci-Algorismia/files/2102933/Examen_8.pdf).
